Spanner/src/Session/CacheSessionPool.php

```diff
@@ -716,11 +723,19 @@ private function createSessions($count)
      */
     private function isSessionValid(array $session)
     {
-        $halfHourBeforeExpiration = $session['expiration'] - (SessionPoolInterface::SESSION_EXPIRATION_SECONDS / 2);
+        $halfHourBeforeExpiration = $session['expiration'] - 1800;
 
+        // sessions more than 28 days old are auto deleted by server
+        if (self::DURATION_SESSION_LIFETIME + $session['creation'] <
+            $this->time() + SessionPoolInterface::SESSION_EXPIRATION_SECONDS) {
+            return false;
+        }
```

Comment:
nit, but I still think this block is unnecessary - Why not just let the 28-hour session get cleaned up in `maintain`, and let the normal expiration check handle them here? That would mean that you could potentially have a session go 28 days + 30 minutes, but I assume that's harmless.

Reply:
There's a chance a session is in `inUse` while maintain was running. Since it was not in `queue`, it never gets cleared even after it is expired and it will always throw errors, unless the `$database->close()` is called.

This check prevents a session from getting used if it's released even once in its last hour of lifetime.